source/common/access_log/grpc_access_log_impl.cc

```diff
@@ -162,22 +159,74 @@ void HttpGrpcAccessLog::log(const Http::HeaderMap* request_headers,
         *common_properties->mutable_downstream_local_address());
   }
   common_properties->mutable_start_time()->MergeFrom(
-      Protobuf::util::TimeUtil::MicrosecondsToTimestamp(
-          std::chrono::duration_cast<std::chrono::microseconds>(
+      Protobuf::util::TimeUtil::NanosecondsToTimestamp(
+          std::chrono::duration_cast<std::chrono::nanoseconds>(
               request_info.startTime().time_since_epoch())
               .count()));
-  if (request_info.requestReceivedDuration().valid()) {
+
+  if (request_info.lastDownstreamRxByteReceived().valid()) {
     common_properties->mutable_time_to_last_rx_byte()->MergeFrom(
-        Protobuf::util::TimeUtil::MicrosecondsToDuration(
-            request_info.requestReceivedDuration().value().count()));
+        Protobuf::util::TimeUtil::NanosecondsToDuration(
+            RequestInfo::TimingUtils::duration<std::chrono::nanoseconds>(
```

Comment:
Please add a helper function for the common part:
```
void helper(some_type& dest, const Optional<MonotonicTime>& src)
   if (src.valid()) {
       dst.MergeFrom(
          Protobuf::util::TimeUtil::NanosecondsToDuration(
             RequestInfo::TimingUtils::duration<std::chrono::nanoseconds>(
                 request_info, src)
                 .value()
                 .count()));
```

Or similar

Reply:
by calling `mutable_*`, it actually creates an empty message for that field, so this particular form doesn't work. With the switch to the RequestInfo returning nanos, however, it's pretty DRY'd up though.